### PR TITLE
fix ENCORE compilation warnings

### LINKED ISSUE #1390: type mismatch (long vs float) in CAffinityPropagation
### Expected behaviour
No compiler warning about type mismatch in CAffinityPropagation function declaration and definition.

### Actual behaviour
```
[...]
gcc -pthread -fno-strict-aliasing -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m32 -march=i686 -fasynchronous-unwind-tables -D_GNU_SOURCE -fPIC -fwrapv -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m32 -march=i686 -fasynchronous-unwind-tables -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m32 -march=i686 -fasynchronous-unwind-tables -fPIC -I/usr/lib/python2.7/site-packages/numpy/core/include -IMDAnalysis/analysis/encore/clustering/include -I/usr/include/python2.7 -c MDAnalysis/analysis/encore/clustering/affinityprop.c -o build/temp.linux-i686-2.7/MDAnalysis/analysis/encore/clustering/affinityprop.o -O3 -ffast-math -std=c99
In file included from /usr/lib/python2.7/site-packages/numpy/core/include/numpy/ndarraytypes.h:1809:0,
                 from /usr/lib/python2.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:18,
                 from /usr/lib/python2.7/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                 from MDAnalysis/analysis/encore/clustering/affinityprop.c:460:
/usr/lib/python2.7/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning "Using deprec
ated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
 #warning "Using deprecated NumPy API, disable it by " \
  ^~~~~~~
MDAnalysis/analysis/encore/clustering/affinityprop.c: In function '__pyx_pf_10MDAnalysis_8analysis_6encore_10clustering_12affinityprop_19AffinityPropagation_run':
MDAnalysis/analysis/encore/clustering/affinityprop.c:2221:180: warning: passing argument 7 of 'CAffinityPropagation' from incompatible pointer type [-Wincompatible-pointer-types]
   __pyx_t_10 = __Pyx_PyInt_From_int(CAffinityPropagation(((float *)__pyx_v_matndarray->data), __pyx_v_cn, __pyx_v_lam, __pyx_v_max_iterations, __pyx_v_convergence, __pyx_v_noise, ((long *)__pyx_v_clusters->data))); if (unlikely(!__pyx_t_10)) __PYX_ERR(0, 104, __pyx_L1_error)
                                                                                                                                                                                    ^
In file included from MDAnalysis/analysis/encore/clustering/affinityprop.c:465:0:
MDAnalysis/analysis/encore/clustering/include/ap.h:30:5: note: expected 'float *' but argument is of type 'long int *'
 int CAffinityPropagation(float*, int, float, int, int, int, float*);
     ^~~~~~~~~~~~~~~~~~~~
```

### Code to reproduce the behaviour

``` bash
$ grep -n CAffinityPropagation MDAnalysis/analysis/encore/clustering/{caffinityprop.pxd,include/ap.h} 
MDAnalysis/analysis/encore/clustering/caffinityprop.pxd:42:    int CAffinityPropagation(float*, int, float, int, int, bint, long*)
MDAnalysis/analysis/encore/clustering/include/ap.h:30:int CAffinityPropagation(float*, int, float, int, int, int, float*);
```

I'm not sure which type is correct here, but it seems it's only ever called from `affinityprop.pyx`
and the parameter is converted to `long*` anyway:
```
$ grep -n CAffinityPropagation MDAnalysis/analysis/encore/clustering/affinityprop.pyx
MDAnalysis/analysis/encore/clustering/affinityprop.pyx:104:        iterations = caffinityprop.CAffinityPropagation( <float*>matndarray.data, cn, lam, max_iterations, convergence, noise, <long*>clusters.data)
```

### Currently version of MDAnalysis:
0.16.1

### PR DESCRIPTION
**Pull request by mtiberti**
Fixes #1390 

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c [hunk 1]
```diff
@@ -22,10 +22,12 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-//#include <math.h>
+#include <math.h>
 #include <time.h>
 #include <sys/types.h>
 #include <time.h>
+#include <sys/types.h>
+#include <unistd.h>
```
Comment (kain88-de) [inline package/MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c]: The `sys/types.h` header is repeated twice. See line 27. 

Why are you including `unistd.h`? I don't see you using types defined in that header.

Why include `math.h` again?

File: package/MDAnalysis/analysis/encore/clustering/include/ap.h [hunk 2]
```diff
@@ -27,4 +27,4 @@ float min(float*, int);
 
 float max(float*, int);
 
-int CAffinityPropagation(float*, int, float, int, int, int, float*);
+int CAffinityPropagation(float*, int, float, int, int, int, long*);
```
Comment (kain88-de) [inline package/MDAnalysis/analysis/encore/clustering/include/ap.h]: Does the code calling this know about the long?

File: package/MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c (hunk 1 above)
Comment (mtiberti) [inline package/MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c]: sorry for the double include - I'll remove one

`unistd.h` is necessary for getpid() (line 190)

the inclusion of math.h fixes another warning:

`MDAnalysis/analysis/encore/dimensionality_reduction/src/spe.c:73:12: warning: implicit declaration of function ‘sqrt’ [-Wimplicit-function-declaration]`

File: package/MDAnalysis/analysis/encore/clustering/include/ap.h (hunk 2 above)
Comment (mtiberti) [inline package/MDAnalysis/analysis/encore/clustering/include/ap.h]: yes - it's expected already in `affinityprop.pyx`

**Comment by kain88-de** [comment]
@mtiberti Once you add yourself to the changelog this can be merged. Thanks

**Comment by mtiberti** [comment]
Done. Thanks to you for the review

**Comment by kain88-de** [comment]
@mtiberti I'm just seeing that this is based on an old state of develop. I'm fixing it

**Comment by mtiberti** [comment]
Yup, I just realised it myself - thanks!

**Comment by rathann** [comment]
Is the cast to `long *` still required in [MDAnalysis/analysis/encore/clustering/affinityprop.pyx](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/analysis/encore/clustering/affinityprop.pyx#L104):
``` Python
        iterations = caffinityprop.CAffinityPropagation( <float*>matndarray.data, cn, lam, max_iterations, convergence, noise, <long*>clusters.data)
```
after this?

**Pull request merged**